### DISCUSSION TITLE
How does Mojo's `parallelize` function distribute sub-tasks across CPU cores?

### DISCUSSION THREAD

**Discussion by JackKelly** [modular/modular#1477] [Q&A]
Does [`parallelize`](https://docs.modular.com/mojo/stdlib/algorithm/functional.html#parallelize) spin up a pool of worker threads, one thread per CPU core, and use work stealing to balance the tasks across threads? (like [Rayon](https://github.com/rayon-rs/rayon/blob/master/FAQ.md#how-does-rayon-balance-work-between-threads)?)

Or does `parallelize` spin up one OS thread per task? Or use a blocking queue to distribute tasks to a thread pool?

**Comment by Brian-M-J** [comment]
I would also really like to know about a lot of details of Mojo's internals. I've heard the team mention their ambitions about the Actor model. Maybe each task is implemented as an actor internally.
